app/src/main/java/org/mozilla/tv/firefox/pinnedtile/PinnedTileAdapter.kt

```diff
@@ -95,49 +98,39 @@ class PinnedTileAdapter(
         iconView.layoutParams = layoutMarginParams
     }
 
-    /**
-     * takes in the home tiles cache and updates the adapter's data source
-     * and UI accordingly, assuming only one new tile is added
-     */
-    fun updateAdapterSingleInsertion(pinnedTiles: MutableList<PinnedTile>) {
-        if (pinnedTiles.size == tiles.size) {
-            // The lists must not be the same size in order
-            // for an insertion to be valid
+    fun setTiles(newTiles: MutableList<PinnedTile>) {
+        if (itemCount == 0) {
```

Comment:
Why don't we want DiffUtil to handle this state too?

Reply:
For some odd android reason, `notifyDataSetChanged()` is specifically required to refresh the entire list when adding a new pinned item to an empty adapter. I'm not too sure why DiffUtil doesn't handle this